components/eamxx/src/physics/shoc/eamxx_shoc_process_interface.cpp

```diff
@@ -85,6 +85,9 @@ void SHOCMacrophysics::set_grids(const std::shared_ptr<const GridsManager> grids
   // Output variables
   add_field<Computed>("pbl_height",    scalar2d    , m,            grid_name);
   add_field<Computed>("inv_qc_relvar", scalar3d_mid, pow(kg/kg,2), grid_name, ps);
+  add_field<Computed>("eddy_diff_heat",   scalar3d_mid, m2/s,        grid_name, ps);
+  add_field<Computed>("w_variance",       scalar3d_mid, m2/s2,       grid_name, ps);
+  add_field<Computed>("cldfrac_liq_prev", scalar3d_mid, nondim,      grid_name, ps);
```

Comment:
I agree, it will be great if we can get these quantities using a function with some inputs rather than adding code in other processes. It makes the logic simpler. 
In ACI, we need `tke` and we derive it from `w_sec_`. We should probably use the `tke` directly as it is one of SHOC's outputs. I won't change the code at this time as it will trigger another cycle of evaluations. But I will do some side calculations to see how close the results are and later use `tke` directly if we can.

Reply:
ACI codes need `tke` at the interfaces. I think SHOC has an internal `tke` at the interfaces.

Reply:
If the prev field is only exposed b/c it's needed for restarts, then it should be created separately, and declared as internal field. You can see an example in [homme](https://github.com/E3SM-Project/scream/blob/6557feb759786a7ed602a22d350e16bdca11ef77/components/eamxx/src/dynamics/homme/eamxx_homme_process_interface.cpp#L241).

Reply:
NOTE: internal fields are treated by the AD as follows:
- should not be used by other atm procs (errors out if they are)
- are added to the RESTART group, so that they are written to restart file
- they are not needed for normal initialization (not looked for in the IC file).

Reply:
Ah, I didn't realize what's happening. Here we have SHOC computing a field just for the sake of ACI. That is indeed underwhelming. Can't ACI compute this field?

Reply:
ACI needs to know the values of liquid cloud fraction before SHOC alters it. The `_prev` field is not added for the restarts, it is added because ACI needs it.

`w_variance` is already computed by SHOC. I just exposed it as ACI codes need to use it. I mimicked what was being done on EAMv2. In EAMv2, CLUBB computes w_variance and the ACI codes use it. We can revisit it as we may be able to use the `tke` directly in the ACI codes.

Reply:
I don't think using tendencies is a good approach, as it requires one to remember adding the tendency request to shoc from input file (besides adding more rounding errors).

Looking at mam4xx's `dropmixnuc` inline docs, I see that the variable `cldo` (the _prev var here) is "cloud fraction on previous time step". I'm a bit confused then: is it "from the previous timestep" or "whatevet was before shoc"? If the latter, why does ACI needs to know the value right before shoc? Can you explain what's the rationale behind it?

I want to try to avoid creating teleconnections between different packages, so understanding the underlying reason may help find a better way...

Reply:
The EAGLES team in planning to look into this issue in more detail, but their top priority right now is to simply get an implementation of the original mam4xx code running.  My two cents: I think it's going to be very hard to test whether changes to the treatment of vertical velocity variances are improvements (in terms of model behavior) until the full aerosol life cycle is implemented and longer tests can be performed... so, kicking the can down the road seems like a reasonable approach here to me

Reply:
>Looking at mam4xx's dropmixnuc inline docs, I see that the variable cldo (the _prev var here) is "cloud fraction on previous time step". I'm a bit confused then: is it "from the previous timestep" or "whatevet was before shoc"? If the latter, why does ACI needs to know the value right before shoc? Can you explain what's the rationale behind it?

Here is an explanation by @kaizhangpnl about the use of this field:
```
liqcldfo and liqcldf are used to determine whether there is growing or shrinking 
cloud in the grid box. liqcldfo is the cloud fraction before CLUBB updates it - same 
as the old cloud fraction used for cloud microphysics and radiation in the previous 
time step (after CLUBB update).
```

Reply:
We will remove this field from the SHOC and compute it in the ACI interface in a follow-up PR.